lib/options.nix

```
@@ -79,55 +138,39 @@ rec {
   defaultNullOpts =
     let
       # Convert `defaultNullOpts`-style arguments into normal `mkOption`-style arguments,
-      # i.e. moves `default` into `description` using `defaultNullOpts.mkDesc`
+      # i.e. merge `default` or `defaultText` into `defaultText`.
       #
-      # "Plugin default" is only added if `args` has a `default` attribute
+      # "Plugin default" is only added if `args` has either a `default` or `defaultText` attribute.
       convertArgs =
         args:
         (
-          args
+          # TODO filter pluginDefault
+          (filterAttrs (
+            n: _:
+            !(elem n [
+              "pluginDefault"
+              "defaultText"
+            ])
+          ) args)
```

Once `defaultNullOpts` migrates `default` -> `pluginDefault`, this can be simplified:

```suggestion
          assert !(args ? default || args ? defaultText);
          (filterAttrs (n: _: n != "pluginDefault") args)
```